### ISSUE TITLE
Cover expression-bodied async methods with generic Task-like return values in implicit object creation analyzer

### ISSUE THREAD

**Issue by John-Leitch** [dotnet/roslyn#77670]
**Brief description:**

Expression-bodied constructs with the `async` modifier and a generic Task-like return type e.g. `Task<T>` should be covered by the implicit object creation analyzer as the type used to close the generic is generally recognized as the return type of the function.

**Languages applicable:**

C#

**Code example that the analyzer should report:**

```cs
using System.Threading.Tasks;

class C
{
    async Task<C> Func() => new [|C|]();
}
```

**Additional information:**

Encountered while using the analyzer, and was so unexpected I thought the analyzer had broken.

**Documentation requirements:**

Shouldn't have significant impact.
